### ISSUE TITLE
[plugin-node-resolve] Can we forcefully exclude `devDependencies`?

### ISSUE THREAD

**Issue by albertodiazdorado** [rollup/plugins#1617]
- Rollup Plugin Name: plugin-node-resolve
- Rollup Plugin Version: 15.2.3

### Feature Use Case
I would like to prevent that I include any `devDependency` in my bundle by accident. I could achieve this by pruning my dev dependencies before going to the `build` step (e.g. `npm prune --production && npm run build`), but I'd find it more ergonomic if we could pass a flag to `node-resolve` to let it know about it.

The reason is that I would like to exclude my `devDependencies` from the licenses & vulnerabilities analysis, but I only may do that if I am certain that they will not be part of the production bundle.

If there is any other means to achieve this that I am not aware of, please let me know :)

### Feature Proposal
```js
plugins: [nodeResolve({ excludeDevDependencies: true })]
```
I do not even know whether this is feasible. If it is, I would be willing to implement it myself.

**Comment by albertodiazdorado** [comment]
@sgpinkus if you require a dev dependency by mistake, it will be included in the bundle

**Comment by sgpinkus** [comment]
That's strange. If you require a dev dependency then presumably you need it.

FYI I came across this issue because I had plugin-node-resolve including some really strange devDeps from deps very deep in the dep graph that I definitely wasn't using and couldn't figure out why they would be included.

**Comment by albertodiazdorado** [comment]
> If you require a dev dependency then presumably you need it.

Yes! Either you stop requiring it, or you install it as a direct dependency. In either case I'd love it if `plugin-node-resolve` would throw an error instead of happily bundling it.
